backend/bittan/bittan/tests/views/start_payment/test_start_payment.py

```diff
@@ -0,0 +1,158 @@
+        
+        test_ticket = TicketType.objects.create(price=200, title="Test Ticket", description="A ticket for testing.")
+        self.test_event.ticket_types.add(test_ticket)
+        
+        test_ticket2 = TicketType.objects.create(price=100, title="Test Ticket 2", description="A ticket for testing number 2.")
+        self.test_event.ticket_types.add(test_ticket2)
+
+        secret_ticket = TicketType.objects.create(price=0, title="Secret Ticket", description="A free ticket (very secret)", is_visible=False)
+        self.test_event.ticket_types.add(secret_ticket)
+
+        self.client = Client()
+
+        self.reservation_response = self.client.post(
+            "/reserve_ticket/", 
+            {
+                "chapter_event": str(self.test_event.pk),
+                "tickets": [
+                    {
+                        "ticket_type": "Test Ticket",
+                        "count": 1
+                        }
+                    ]
+            },
+            content_type="application/json"
+        )
+
+    def test_start_payment(self):
+        mail_address = "mail@mail.com"
+        response = self.client.post(
+            "/start_payment/",
+            {
+                "email_address": mail_address
+            }
+        )
+
+        self.assertEqual(response.status_code, 200)
+        payment_id = self.client.session["reserved_payment"]
+        payment = Payment.objects.get(pk=payment_id)
+        self.assertEqual(payment.payment_started, True)
+        self.assertEqual(payment.email, mail_address)
+        self.assertEqual(payment.status, PaymentStatus.RESERVED)
+    
+    def test_invalid_mail(self):
+        response = self.client.post(
+            "/start_payment/",
+            {
+                "email_address": "dsjklasdfljka"
+            }
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_invalid_session_token(self):
+        new_client = Client()
+        response = new_client.post(
+            "/start_payment/",
+            {
+                "email_address": "mail@mail.com"
+            }
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_expired_session_out_of_tickets(self):
+        payment_id = self.client.session["reserved_payment"]
+        payment = Payment.objects.get(pk=payment_id)
+        payment.status = PaymentStatus.FAILED_EXPIRED_RESERVATION
+        payment.save()
+
+        client2 = Client()
+        _ = client2.post(
+            "/reserve_ticket/", 
+            {
+                "chapter_event": str(self.test_event.pk),
+                "tickets": [
+                    {
+                        "ticket_type": "Test Ticket",
+                        "count": 10
+                        }
+                    ]
+            },
+            content_type="application/json"
+        )
+
```

Comment:
Consider checking that this request was successful

Reply:
Solved as of ee6e07861dc7259afa5bc766e14c8684b566d9b9.